### PR TITLE
feat: added declaration file for event builder index.d.ts

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
- Added declaration file under optimizely-sdk/lib/core/event_builder/index.d.ts 

## Test plan
- All unit tests must work fine.

### CODE REVIEW

**Comment by msohailhussain** [comment]
This PR is based on project config declaration branch, before merging into master, we'll have to rebase once project config declaration file PR is merged.

File: packages/optimizely-sdk/lib/core/event_builder/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     * @return {EventLoggingEndpoint} Params to be used in conversion event logging endpoint call
+     */
+    getConversionEvent(options: ConversionEventOptions): EventLoggingEndpoint;
+
+    /**
+     * Creates an ImpressionEvent object from decision data
+     * @param {ImpressionConfig} config
+     * @return {ImpressionEvent} an ImpressionEvent object
+     */
+    buildImpressionEvent(config: ImpressionConfig): ImpressionEvent;
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/core/event_builder/index.d.ts:91]: I think `buildConversionEvent` and `buildImpressionEvent` should live in its own `event_helpers.d.ts` file as these methods are imported separately from `event_builder`

Comment (msohailhussain) [reply packages/optimizely-sdk/lib/core/event_builder/index.d.ts:91]: yeah you are right, i missed it.

File: packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare module '@optimizely/optimizely-sdk/lib/core/event_builder' {
+  import { ProjectConfig } from '@optimizely/optimizely-sdk/lib/core/project_config';
+  import { EventTags, UserAttributes } from '@optimizely/optimizely-sdk';
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts:19]: please rebase and update these imports. `UserAttributes` is now live in `shared_types.ts`. EventTags can also be moved there and be imported here.

File: packages/optimizely-sdk/lib/core/event_builder/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare module '@optimizely/optimizely-sdk/lib/core/event_builder' {
+  import { ProjectConfig } from '@optimizely/optimizely-sdk/lib/core/project_config';
+  import { LogHandler } from '@optimizely/js-sdk-logging';
+  import { EventTags, UserAttributes } from '@optimizely/optimizely-sdk';
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/core/event_builder/index.d.ts:20]: Same comment as above.

File: packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    variationKey: string;
+    userId: string;
+    userAttributes: UserAttributes;
+    clientEngine: string;
+    clientVersion: string;
+    configObj: ProjectConfig;
+  }
+
+  // eslint-disable-next-line  @typescript-eslint/no-empty-interface
+  interface ImpressionEvent {}
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts:32]: It seems wrong to keep these empty, presumably the caller would do something with it. Even if it's passing it on to event processor. The event processor exposes type definitions, so we should be able to match or use those.

File: packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts (hunk 2 above)
Comment (mjc1283) [reply packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts:19]: Agree, please make this change.

File: packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  // eslint-disable-next-line  @typescript-eslint/no-empty-interface
+  interface ConversionEvent {}
+
+  export interface EventHelpers {
+    /**
+     * Creates an ImpressionEvent object from decision data
+     * @param {ImpressionConfig} config
+     * @return {ImpressionEvent} an ImpressionEvent object
+     */
+    buildImpressionEvent(config: ImpressionConfig): ImpressionEvent;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/event_builder/event_helpers.d.ts:53]: These functions should be exported directly from this module, not within an `EventHelpers` interface

File: packages/optimizely-sdk/lib/core/event_builder/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    eventTags: EventTags;
+  }
+
+  export interface EventBuilder {
+    /**
+     * Create impression event params to be sent to the logging endpoint
+     * @param  {ImpressionOptions} options Object containing values needed to build impression event
+     * @return {EventLoggingEndpoint} Params to be used in impression event logging endpoint call
+     */
+    getImpressionEvent(options: ImpressionOptions): EventLoggingEndpoint;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/event_builder/index.d.ts:52]: These functions should be exported directly from this module, not within an `EventBuilder` interface

**Review by mjc1283 — CHANGES_REQUESTED** [review]

**Review by mjc1283 — APPROVED** [review]

File: packages/optimizely-sdk/lib/core/event_builder/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ * limitations under the License.
+ */
+
+declare module '@optimizely/optimizely-sdk/lib/core/event_builder' {
+  import { ProjectConfig } from '@optimizely/optimizely-sdk/lib/core/project_config';
+  import { LogHandler } from '@optimizely/js-sdk-logging';
+  import { Event as EventLoggingEndpoint } from '@optimizely/optimizely-sdk';
+
+  interface ImpressionOptions {
+    attributes: import('../../shared_types').UserAttributes;
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/core/event_builder/index.d.ts:23]: `attributes`, `eventKey`, and `logger` are optional attributes. please add `?` to it:
```
attributes?: import('../../shared_types').UserAttributes;
eventKey?: eventKey;
logger?: LogHandler;
```

File: packages/optimizely-sdk/lib/core/event_builder/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    configObj: ProjectConfig;
+    experimentId: string;
+    eventKey: string;
+    variationId: string;
+    logger: LogHandler;
+    userId: string;
+  }
+
+  interface ConversionEventOptions {
+    attributes: import('../../shared_types').UserAttributes;
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/core/event_builder/index.d.ts:35]: `attributes` and `eventTags` are optional attributes.

**Review by yavorona — CHANGES_REQUESTED** [review]
Looks great except for a couple of minor comments. @mjc1283 please give :+1:   if you agree

**Review by yavorona — APPROVED** [review]

**Pull request merged**